File: Python/Flappy Bird/bird.py
class Bird:
    def __init__(self, x, y, radius, dy = 0) -> None:
        self.x = x
        self.y = y
        self.radius = radius
        self.dy = dy

    def jump(self, ay) -> None:
        self.dy = ay

    def update(self, g = 0.1) -> None:
        self.gravity(g)
        self.y += self.dy

    def gravity(self, g):
        self.dy += g

File: JavaScript/Flappy Bird (NN)/bird.js
class Bird {
    constructor(x, y, gravity) {
        this.x = x
        this.y = y
        this.gravity = gravity
        this.dy = -5
        this.died = false
        this.score = 0

        this.brain = new NeuralNetwork(4, 4, 1)
    }

    think(pipes) {
        let proxPipe
        for (let i = 0; i < pipes.length; i++) {
            if (pipes[i].proximo) {
                proxPipe = pipes[i]
                break
            }
        }

        let inputs = []
        inputs[0] = this.y / height
        inputs[1] = proxPipe.buracoY / height
        inputs[2] = (proxPipe.buracoY + proxPipe.alturaBuraco) / height
        inputs[3] = proxPipe.pipeX / width

        inputs = tf.tensor2d([inputs])

        let output = this.brain.predict(inputs).arraySync()[0][0];

        if (output > 0.5) this.jump()
    }

    update() {
        this.dy += this.gravity
        this.y += this.dy
    }

    jump() {
        this.dy = -birdJump
    }
}
